cdap-docs/developers-manual/source/data-exploration/streams.rst

```diff
@@ -0,0 +1,214 @@
+with a few differences:
+ 
+  * Map keys do not have to be strings, but can be any type.
+  * No "name" property for enum type. 
+  * No support of "doc" and "aliases" in record and enum types.
+  * No support of "doc" and "default" in record field.
+  * No "fixed" type.
+
+There are a few additional limitations on the types of schemas that can be used for exploration: 
+ 
+  * Schemas must be a record of at least one field.
+  * Enums are not supported.
+  * Unions are not supported, unless it is a union of a null and another type, representing a nullable type.
+  * Recursive types are not supported. This means you cannot have a record field that references itself in one of its fields.
+
+Data exploration using Impala has additional limitations:
+
+  * Fields must be a scalar type (no maps, arrays, or records)
+
+On top of these general limitations, each format has its own restrictions on the types
+of schemas they support. For example, the CSV format does not support maps or records as
+data types. Please see the javadocs for each format for more information about the types
+of schemas they support. 
+
+Schema Syntax
+-------------
+Schemas are represented as JSON Objects, following the same format as Avro schemas
+`<http://avro.apache.org/docs/1.7.3/spec.html#schemas>`__. 
+The JSON representation is used by the RESTful APIs.
+The CLI also supports a SQL-like syntax. 
+
+For example, the SQL-like schema::
+
+  f1 int, f2 string, f3 array<int> not null, f4 map<string, int> not null, f5 record<x:int, y:double> not null
+
+is equivalent to the Avro-like JSON schema::
+
+  {
+    "type": "record",
+    "name": "rec",
+    "fields": [
+      { 
+        "name": "f1",
+        "type": [ "int", "null" ]
+      },
+      { 
+        "name": "f2",
+        "type": [ "string", "null" ]
+      },
+      { 
+        "name": "f3",
+        "type": { "type": "array", "items": [ "int", "null" ] } 
+      },
+      { 
+        "name": "f4", 
+        "type": { 
+          "type": "map", "keys": [ "string", "null" ], "values": [ "int", "null" ] 
+        }
+      },
+      { 
+        "name": "f5",
+        "type": {
+          "type": "record",
+          "name": "rec1",
+          "fields": [
+            { "name": "x", "type": [ "int", "null" ] },
+            { "name": "y", "type": [ "double", "null" ] }
+          ]
+        }
+      }
+    ]
+  }
+
+Text Format
+-----------
+The ``text`` format simply interprets each event body as a string. The format supports a very limited
+schema, namely a record with just one field of type ``string``. The format supports a ``charset`` setting
+that allows you to specify the charset of the text. It defaults to ``utf-8``.
+
+For example::
+
+  set stream format mystream text "data string not null" "charset=ISO-8859-1"
+
+CSV and TSV Formats
+-------------------
+The ``csv`` and ``tsv`` formats read event bodies as delimited text.
```

Comment:
The ``csv`` (comma separated values) and ``tsv`` (tab separated values) formats read event bodies as delimited text.